torch/_C/__init__.pyi.in

```diff
@@ -74,6 +76,8 @@ P = ParamSpec("P")  # noqa: PYI001
 R = TypeVar("R", covariant=True)  # return value (always covariant)  # noqa: PYI001
 T_co = TypeVar("T_co", covariant=True)  # noqa: PYI001
 
+NestedList: TypeAlias = Union[Number, _complex, _bool, list['NestedList']]
```

Comment:
This is just dealing with the `Tensor.tolist` method mentioned above - it needs to be recursive, and so I have updated the template file with this type alias to handle this.